integrations/java/iceberg-1.2/openhouse-java-runtime/src/main/java/com/linkedin/openhouse/javaclient/OpenHouseTableOperations.java

```diff
@@ -166,13 +167,38 @@ protected CreateUpdateTableRequestBody constructMetadataRequestBody(
     // set tableType from incoming metadata to createUpdateTableRequestBody
     if (metadata.properties() != null
         && metadata.properties().containsKey(OPENHOUSE_TABLE_TYPE_KEY)) {
-      createUpdateTableRequestBody.setTableType(
-          CreateUpdateTableRequestBody.TableTypeEnum.valueOf(
-              metadata.properties().get(OPENHOUSE_TABLE_TYPE_KEY)));
+      createUpdateTableRequestBody.setTableType(getTableType(base, metadata));
     }
     return createUpdateTableRequestBody;
   }
 
+  // If request is coming from replication process, createUpdateTableRequestBody.tableType should be
+  // REPLICA_TABLE
+  // Replication process requests are identified based on difference between table types and
+  // cluster_id between base,
+  // metadata
+  @VisibleForTesting
+  CreateUpdateTableRequestBody.TableTypeEnum getTableType(
+      TableMetadata base, TableMetadata metadata) {
+    if (base != null) {
+      CreateUpdateTableRequestBody.TableTypeEnum baseTableType =
+          CreateUpdateTableRequestBody.TableTypeEnum.valueOf(
+              base.properties().get(OPENHOUSE_TABLE_TYPE_KEY));
+      CreateUpdateTableRequestBody.TableTypeEnum metadataTableType =
+          CreateUpdateTableRequestBody.TableTypeEnum.valueOf(
+              metadata.properties().get(OPENHOUSE_TABLE_TYPE_KEY));
```

Comment:
small readability nit this can be moved out of the if statement since we are returning the `metadataTableType` as default if the table is not being replicated